addon/-private/system/model/internal-model.js

```diff
@@ -348,6 +350,38 @@ export default class InternalModel {
         isError: this.isError,
         adapterError: this.error
       };
+      let hasCreateProperties = typeof properties === 'object' && properties !== null;
+      let initializedRelationships;
+
+      if (hasCreateProperties) {
+        initializedRelationships = [];
+
+        if ('id' in properties) {
+          this.setId(properties.id);
+          delete properties.id;
+        }
+
+        this.eachAttribute((name) => {
+          if (name in properties) {
+            this.setDirtyAttribute(name, properties[name]);
+            delete properties[name];
+          }
+        });
+
+        this.eachRelationship((name, { kind }) => {
+          if (name in properties) {
+            initializedRelationships.push(name);
+            if (kind === 'belongsTo') {
+              this.setDirtyBelongsTo(name, properties[name]);
+            } else {
+              this.setDirtyHasMany(name, properties[name]);
+            }
+            delete properties[name];
+          }
+        });
+
+        emberAssign(createOptions, properties);
```

Comment:
I would prefer to avoid mutating the provided POJO (mutating for example, completely breaks using a shared POJO to .createRecord across many tests). A straightforward way to do this would be to invert this method a bit:

```js
if (hasCreateProperties) {
  let classFields = this.getFields();

  for (let propertyName in properties) {
    let propertyValue = properties[propertyName];
    let fieldType = classFields.get(propertyName);

    if (propertyName === 'id') {
      this.setId(properties.id);
      continue;
    }

    switch (fieldType) {
      case 'attribute':
        this.setDirtyAttribute(name, propertyValue);
        break;
      case 'belongsTo':
        this.setDirtyBelongsTo(name, propertyValue);
        initializedRelationships.push(name);
        break;
      case 'hasMany':
        this.setDirtyHasMany(name, propertyValue);
        initializedRelationships.push(name);
        break;
      default:
        createOptions[propertyName] = propertyValue;
    }
  }
}


// ...snip...

getFields(attributeName) {
  return get(this.modelClass, 'fields');
}
```